GitHub Actions Workflow to download and pre-process CHIRPS-GEFS

zackarno opened the pull request:
This work-flow is running successfully when manually run. It does the folliwng:

- download the latest CHIRPS-GEFS leadtimes 1-10 rasters into memory. 
- It then crops these to the area of interest. The cropped rasters are held in memory for processing and also downloaded to the specified gdrive directory.
-  zonal mean is calculated per governorate for each leadtime on the in memory rasters. The resulting object is held in memory for next processing steps, but also downloaded to specified gdrive directory as a csv.

zackarno committed:
script install req packages

zackarno committed:
download chirps gefs (global)-> clip to AOI -> upload clipped tifs to gdrive -> run zonal stats for AOI

zackarno committed:
script to update trigger data

zackarno committed:
change `tidyverse` to `dplyr`

zackarno committed:
try rm `exactextractr` - see if `terra::crop` will run.

zackarno committed:
rm exactextractr from trigger run script

zackarno committed:
try specific version of `{googledrive}`

zackarno committed:
include readr - updated to `{googledrive 2.1.0}` on local and still working locally

zackarno committed:
move env vars to Run yem-trigger-monitoring job

zackarno committed:
include `{purrr}`

zackarno committed:
try switching to ubuntu

zackarno committed:
change R version for ubuntu run

zackarno committed:
add dep line for linux

zackarno committed:
v2

zackarno committed:
try adding gdal deps for linux

zackarno committed:
try catching R packages based on typhoon repo yaml

zackarno committed:
add `{exactextractr}` back

zackarno committed:
change run date to yesterday since gefs wont be released for 3 hours

zackarno committed:
add `tidyr` for `pivot_wider`

zackarno committed:
sourcing req function added back in after accidental removal. Start roxy for func

zackarno committed:
clean up documentation/roxygen and set CRON time for 910 AM utc

zackarno committed:
add req packages for auto-email

zackarno committed:
adj pipeline so gdrive locations are inputs into `load_chirps_gefs()`

zackarno committed:
basic email template working

zackarno committed:
adj typo in CRON time

zackarno committed:
tmp changes to shorten workflow to see if workflow runs on GH actions

zackarno committed:
change `gghdx()` GH url

zackarno committed:
add GITHUB_PAT to yaml and change order of package installation

zackarno committed:
change order of package installation again.

zackarno committed:
`all_drib` -> `drive_dribble`

turnerm commented on .github/workflows/yem-trigger-monitoring.yaml:
```diff
@@ -6,34 +6,48 @@ name: yem-trigger-monitoring
   workflow_dispatch:
   schedule:
-    - cron: '25 12 * * *'
+    - cron: '10 09 * * *' # chirps-gefs should be updated 9AM UTC, let's give an extra 10 min
```
Ah did we confirm that the timestamps [here](https://data.chc.ucsb.edu/products/EWX/data/forecasts/CHIRPS-GEFS_precip_v12/daily_16day/2023/05/23/) are PST and not UTC?

zackarno replied:
yes they are PST

turnerm commented on .github/workflows/yem-trigger-monitoring.yaml:
```diff
@@ -6,34 +6,48 @@ name: yem-trigger-monitoring
+          sudo apt-get install libgdal-dev libproj-dev libgeos-dev libudunits2-dev
+      - name: Install dependencies
+        run: |
+          sudo apt-get install -y libxml2-dev # Add any additional dependencies here
```
I would combine the `apt-get installs`, and also start with an `apt-get update`, like [here](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#L48)

turnerm replied:
Do you not need [this](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#LL47C1-L47C1) PPA for GDAL etc?

zackarno replied:
> I would combine the `apt-get installs`, and also start with an `apt-get update`, like [here](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#L48)

good idea - did it here:  dc96d67d7a19123560195f7da57e35832d4b0ae4

> Do you not need [this](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#LL47C1-L47C1) PPA for GDAL etc?

apparently not it seems to be running without it

turnerm commented on .github/workflows/yem-trigger-monitoring.yaml:
```diff
@@ -6,34 +6,48 @@ name: yem-trigger-monitoring
-          source(file.path("src", "update_trigger.R")) 
-        shell: Rscript {0} 
\ No newline at end of file
+          CHD_DS_PORT: ${{ secrets.CHD_DS_PORT }} 
```
If you use pre-commit it will fix these newline issues for you automatically

turnerm reviewed (commented):
Looks good, just a couple of comments around the package installs. (I haven't looked at the R bits, saving that for @PaulineNimo).

But just wondering have you tested it and it works? Without the addition GIS PPA?

zackarno committed:
update date

zackarno committed:
add pandoc action

zackarno committed:
try `here()` to render email

zackarno committed:
rm `file.path()` for `email.rmd`

zackarno committed:
Update update_trigger.R

Add print debug

zackarno committed:
change `.rmd` to `.Rmd` - add `file.exists()` debug print statements

zackarno committed:
tmp change to date to download rasters (for testing) and update yaml

(https://github.com/OCHA-DAP/pa-aa-yem-flooding/pull/24#discussion_r1202161118)

zackarno committed:
update comment on CRON timing. Set download/process func to get latest 1-10 LT for today

zackarno committed:
Merge pull request #25 from OCHA-DAP/auto-plot-latest-gefs

Auto plot latest gefs

zackarno commented:
> Looks good, just a couple of comments around the package installs. (I haven't looked at the R bits, saving that for ? @PaulineNimo).

@PaulineNimo 

The important R components are: 

-  `update_trigger.R` - main script activated by GH actions which kicks off the process
- `load_chirps_gefs.R` - run inside update_trigger.R to deal with CHIRPS gefs : loads -> crops->downloads -> zonal stats
-  `email.Rmd` -  rendered from update_trigger.R to generate an html email
- `.depends.R` - also activated by GH actions directly (gets req packages)

zackarno committed:
run styler, rm unused/ghost comments

zackarno committed:
rm unneeded file.exists() debugging statements

zackarno committed:
adjust html/banners to account for feedback received by global monitoring emails

turnerm reviewed (approved):
Great!

zackarno committed:
re-size plots, add chirps-gefs data source, minor wording revisions

zackarno commented:
just added some feedback received on slack in this commit [91e7f3d](https://github.com/OCHA-DAP/pa-aa-yem-flooding/pull/24/commits/91e7f3d8c6e911818255b5782ebc81551f6472e7)
- minor wording adjustments
- resize plots
- added chirps-gefs data source

going to merge now

zackarno committed:
Merge branch 'main' into GHA-load-process-chirps-gefs
